xayn-ai/src/data/mod.rs

```diff
@@ -1,14 +1,18 @@
 pub mod document;
 pub(crate) mod document_data;
 
+use derive_more::From;
 use serde::{Deserialize, Serialize};
+use uuid::Uuid;
 
 use crate::embedding::utils::Embedding;
 
 // Hint: We use this id new-type in FFI so repr(transparent) needs to be kept
 #[repr(transparent)]
-#[derive(Debug, PartialEq, Eq, Hash, Clone, Copy, Serialize, Deserialize, PartialOrd, Ord)]
-pub struct CoiId(pub usize);
+#[derive(
+    Debug, PartialEq, Eq, Hash, Clone, Copy, PartialOrd, Ord, Serialize, Deserialize, From,
+)]
+pub struct CoiId(pub Uuid);
```

Comment:
should we make the inner field private since we have the newly derived `From` impl now?
```suggestion
pub struct CoiId(Uuid);
```

Reply:
yes makes sense to hide it

Reply:
yes makes sense to hide it